packages/cli/config/jest.js

```diff
@@ -262,19 +328,25 @@ async function getRootConfig() {
         testScript?.includes('backstage-cli test') ||
         testScript?.includes('backstage-cli package test');
       if (testScript && isSupportedTestScript) {
-        return await getProjectConfig(projectPath, {
-          displayName: packageData.name,
-        });
+        return cleanProjectConfig(
+          await getProjectConfig(projectPath, {
+            displayName: packageData.name,
+          }),
+        );
       }
 
       return undefined;
     }),
   ).then(cs => cs.filter(Boolean));
 
+  const rootBaseConfig = await getProjectConfig(targetPath, coverageConfig);
+
   return {
-    rootDir: targetPath,
-    projects: configs,
-    ...coverageConfig,
+    ...rootBaseConfig,
+    ...{
+      rootDir: targetPath,
+      projects: configs,
```

Comment:
I'm not 100% sure if this is inline with the comment from @Rugvip: https://github.com/backstage/backstage/issues/15311#issuecomment-1396804199

Reply:
This is the main fix, if we don't keep the custom Jest config and we just return what the function builds then it won't have the custom `reporters`

Reply:
Reading again that comment I think this fix is the fix for the workspace / root project. We could change the "project" level config load to look at its own config only and if not present look at the root one and stop there.

Reply:
Yep I think it's best to switch this over to be explicit tbh. What that means is that each project config would only search for the `"jest"` field in `package.json` in its own directly, while the top-level root config will only search for the `"jest"` field in the root `package.json`. This would completely remove the directory ancestry traversal for the project configs when running in the root project mode. With this structure we should be able to avoid extra config options and warnings.

Reply:
Could this be a good summary?
- When running the command in the workspace project:
   - Look for its own Jest config in its package.json
   - For each 'project' in the workspace search for config in their 'package.json' (however it could cause warnings as settings at this level are not the same as settings on a root level Jest config)

- When running on a sub 'project'
   - Look for its own Jest config in its package.json